src/ray/core_worker/task_manager.h

```diff
@@ -749,28 +740,34 @@ class TaskManager : public TaskFinisherInterface, public TaskResubmissionInterfa
   /// \param attempt_number The attempt number to record the task status change
   /// event. If not specified, the attempt number will be the current attempt number of
   /// the task.
+  ///
+  /// \note This function updates `task_entry` in place. Please only call
```

Comment:
Note that this note is not equivalent to `ABSL_EXCLUSIVE_LOCKS_REQUIRED(mu_)`.

For example, the following code satisfies `ABSL_EXCLUSIVE_LOCKS_REQUIRED(mu_)` but does not satisfy the requirement described in this note.

```
Acquire lock
Retrieve TaskEntry
Release lock
...
Acquire lock
SetTaskStatus
Release lock
```